mtp_api/apps/transaction/permissions.py

```diff
@@ -12,7 +14,12 @@ def has_permission(self, request, view):
 class IsOwnPrison(BasePermission):
 
     def has_permission(self, request, view):
-        return request.user.prisonusermapping.prisons.filter(pk=view.kwargs.get('prison_id')).exists()
+        try:
+            prison_user_mapping = request.user.prisonusermapping
+        except PrisonUserMapping.DoesNotExist:
+            return False
```

Comment:
The only other way is to make an extra query which doesn't seem great.
This actually happen all the time in python, try/except exceptions is not as bad as it seems in other languages.